src/CommunityToolkit.Maui.Core/Views/Popup/PopupExtensions.android.cs

```diff
@@ -186,6 +180,38 @@ public static void SetSize(this Dialog dialog, in IPopup popup, in AView contain
 		}
 
 		container.LayoutParameters = containerLayoutParams;
+
+
+		static void CalculateSizes(IPopup popup, Context context, ref int realWidth, ref int realHeight, ref int realContentWidth, ref int realContentHeight)
+		{
+			ArgumentNullException.ThrowIfNull(popup.Content);
+
+			if (!popup.Size.IsZero)
+			{
+				realWidth = (int)context.ToPixels(popup.Size.Width);
+				realHeight = (int)context.ToPixels(popup.Size.Height);
+			}
+			if (double.IsNaN(popup.Content.Width) && double.IsNaN(popup.Content.Height))
```

Comment:
Is there ever a chance that one of these could be `NaN` and not the other? I am not sure I would expect this scenario to happen but it just struck me as a possibility when reading this statement

Reply:
Good catch, probably an `||` will be safer here. I'll change it asap